mbf_utility/src/robot_information.cpp

```diff
@@ -59,16 +61,33 @@ bool RobotInformation::getRobotPose(geometry_msgs::PoseStamped &robot_pose) cons
   if (!tf_success)
   {
     ROS_ERROR_STREAM("Can not get the robot pose in the global frame. - robot frame: \""
-                         << robot_frame_ << "\"   global frame: \"" << global_frame_ << std::endl);
+                         << robot_frame_ << "\"   global frame: \"" << global_frame_);
     return false;
   }
   return true;
 }
 
-bool RobotInformation::getRobotVelocity(geometry_msgs::TwistStamped &robot_velocity, ros::Duration look_back_duration) const
+bool RobotInformation::getRobotVelocity(geometry_msgs::TwistStamped &robot_velocity) const
 {
-  // TODO implement and filter tf data to compute velocity.
-  return false;
+  nav_msgs::Odometry base_odom;
+  odom_helper_.getOdom(base_odom);
+  if (base_odom.header.stamp.isZero())
+  {
+    ROS_ERROR_STREAM("No odometry messages received; robot velocity unknown");
```

Comment:
I think there is some room for improvement here:
- given that user code worked so far without an `odom` topic being present, this should be a warning rather than an error?
- since this is not expected to spuriously fail / if it fails, it will most likely continue to fail, use a `THROTTLE` version
- add the odom topic to the warning using `base_odom.getOdomTopic()`

On the larger picture, it seems the `RobotInformation` class in `AbstractNavigationServer` used the default parameter value of the odom topic. Would be better to expose this as a ROS parameter (default initialized to `"odom"`)? Then, the warning here could be extended to inform the user to set this parameter. (Maybe could even allow the user to suppress the warning by explicitly setting the odom topic parameter to an empty string)

Reply:
good idea